cpp/include/kvikio/shim/cufile.hpp

```diff
@@ -107,23 +111,22 @@ class cuFileAPI {
 
     // cuFile is supposed to open and close the driver automatically but because of a bug in
     // CUDA 11.8, it sometimes segfault. See <https://github.com/rapidsai/kvikio/issues/159>.
-    CUfileError_t const error = DriverOpen();
-    if (error.err != CU_FILE_SUCCESS) {
-      throw std::runtime_error(std::string{"cuFile error at: "} + __FILE__ + ":" +
-                               KVIKIO_STRINGIFY(__LINE__) + ": " +
-                               cufileop_status_error(error.err));
+    if (!stream_available) {  // The stream API was introduced in CUDA 12.2.
+      driver_open();
```

Comment:
Actually, the version of CTK and cuFile might not match, updated the comment:
```
    // cuFile is supposed to open and close the driver automatically but
    // because of a bug in cuFile v1.4 (CUDA v11.8), it sometimes segfault:
    // <https://github.com/rapidsai/kvikio/issues/159>.
    // We use the stream API as an version indicator of cuFile, it was introduced
    // in cuFile v1.7 (CUDA v12.2).
```

Reply:
Ah, ok.